.github/workflows/python-build-workflow.yml

```diff
@@ -0,0 +1,36 @@
+name: Python package
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install poetry
+          make install
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: lint fix
```

Comment:
The Makefile is set up to do all of the CI steps in one command: `make ci`

Reply:
make ci runs everything right now. My thinking was that by separating out the different stages, we can allow developers to better know which component of the build process is failing

Reply:
Yeah, not a big deal.  Maybe it depends on what the build output looks like between the two options.

In general it's better to put logic like what it means for a build to "pass" in one place and just point to it from other places that run the build.  That way future engineers don't need to grok both the Makefile and the build workflow config.  Easier debugging is a valid reason to break it up though.

Looks like you went ahead and made the change so seems like we can move forward.